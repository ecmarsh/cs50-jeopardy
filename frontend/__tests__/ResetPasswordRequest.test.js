import { mount } from 'enzyme';
import wait from 'waait';
import toJSON from 'enzyme-to-json';
import { MockedProvider } from 'react-apollo/test-utils';
import RequestReset, {
  REQUEST_RESET_MUTATION,
} from '../components/PasswordForgot';

const mock = {
  request: {
    query: REQUEST_RESET_MUTATION,
    variables: { email: 'test@test.com' },
  },
  result: {
    data: { requestReset: { message: 'success', __typename: 'Message' } },
  },
};

describe('Request Password Reset', () => {
  it('renders and matches snapshot', async () => {
    const wrapper = mount(
      <MockedProvider>
        <RequestReset />
      </MockedProvider>
    );
    const resetForm = wrapper.find('form[data-test="reset-form"]');
    expect(toJSON(resetForm)).toMatchSnapshot();
  });
  it('calls the reset mutation', async () => {
    const wrapper = mount(
      <MockedProvider mocks={[mock]}>
        <RequestReset />
      </MockedProvider>
    );
    // simulate inputting an email
    wrapper.find('input').simulate('change', {
      target: { name: 'email', value: 'test@test.com' },
    });
    // simulate form submission
    wrapper.find('form').simulate('submit');
    await wait();
    wrapper.update();
    // expect a reset confirmation message
    expect(wrapper.find('p').text()).toContain(
      'Check your email for a reset link'
    );
  });
});
