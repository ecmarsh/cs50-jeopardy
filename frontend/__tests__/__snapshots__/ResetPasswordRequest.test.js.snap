// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Request Password Reset renders and matches snapshot 1`] = `
<form
  className="Form-qsq7s5-0 bLtDzK"
  data-test="reset-form"
  method="POST"
  onSubmit={[Function]}
>
  <DisplayError
    error={Object {}}
  />
  <h2>
    Password Reset
  </h2>
  <fieldset
    aria-busy={false}
    disabled={false}
  >
    <label
      htmlFor="email-2"
    >
      <input
        id="email-2"
        name="email"
        onChange={[Function]}
        placeholder="Account Email"
        required={true}
        type="email"
        value=""
      />
      <Utilities__TextMuted>
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "Utilities__TextMuted-s11g5i-2",
                "isStatic": false,
                "lastClassName": "bdrBqE",
                "rules": Array [
                  "opacity:0.8;font-size:0.8em;display:inline;opacity:1;transition:opacity 0.3s ease-in-out;",
                  [Function],
                  " ",
                  [Function],
                  " ",
                  [Function],
                  " ",
                  [Function],
                ],
              },
              "displayName": "Utilities__TextMuted",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "Utilities__TextMuted-s11g5i-2",
              "target": "span",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <span
            className="Utilities__TextMuted-s11g5i-2 bdrBqE"
          >
            Account Email
          </span>
        </StyledComponent>
      </Utilities__TextMuted>
    </label>
    <Button
      secondary={true}
      type="submit"
    >
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "Button-sc-1f1w3gq-0",
              "isStatic": false,
              "lastClassName": "dJqAho",
              "rules": Array [
                "font-family:",
                [Function],
                ";color:white;border-radius:0.5rem;text-transform:uppercase;letter-spacing:1px;transition:.2s ease;&:hover{cursor:pointer;}&:focus{cursor:pointer;outline:",
                [Function],
                " auto 3px;}",
                [Function],
                " ",
                [Function],
                " ",
                [Function],
                " ",
                [Function],
                " ",
                [Function],
                " &.button--sound{font-weight:bolder;text-decoration:line-through;&::after{content:\\"❌\\";padding-left:3px;}",
                [Function],
                "}",
                [Function],
              ],
            },
            "displayName": "Button",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "Button-sc-1f1w3gq-0",
            "target": "button",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        secondary={true}
        type="submit"
      >
        <button
          className="Button-sc-1f1w3gq-0 dJqAho"
          type="submit"
        >
          Request Reset
        </button>
      </StyledComponent>
    </Button>
  </fieldset>
</form>
`;
